### ISSUE TITLE
Wrongly detects unencoded HTML entities with one nesting of double quotes and quotes

### ISSUE THREAD

**Issue by sanmai-NL** [43081j/eslint-plugin-lit#74]
> attribute values may not contain unencoded HTML entities, e.g. use `&gt;` instead of `>`

![image](https://user-images.githubusercontent.com/3374183/101502458-419fa000-3971-11eb-9e2f-d8b555e55bf8.png)

While this works:
```js
const test = html`<component-blo test="['test']">Bla</component-blo>`;
```

**Comment by sanmai-NL** [comment]
Another example:
```html
<a href="https://tasks.office.com/test.onmicrosoft.com/NL-NL/Home/Planner#/plantaskboard?groupId=fc6fdef6-3dbb-4155-a4ed-a65e4cdadd12&planId=CAby-9tdR06SEiFFGkHtIJcAB2iv">
Test plan
</a>
```

**Comment by 43081j** [comment]
The double quotes will be because the regex is a bit 'dumb'.

The HTML spec from what i remember defines:

* Single quotes inside a single-quoted attribute must be encoded as `&pos;` (apostrophe)
* Double quotes inside a double-quoted attribute must be encoded as `&quot;` (quote)

So i think the rule just needs tweaking a bit to differentiate between these two cases.

Your second one is because ampersands must also be encoded as per the spec (as `&amp;`).

Spec states:

> The attribute value must delimited by double-quote characters (") before and after the value, and must not contain any double-quote characters or an ambiguous ampersands in between
> To include a double-quote character within the value, either use a character reference (&quot;), or use a single-quoted attribute value instead

and

> The attribute value must delimited by single-quote characters (') before and after the value, and must not contain any single-quote characters or an ambiguous ampersands in between
> To include a single-quote (or apostrophe) character within the value, either use a character reference (&apos;), or use a double-quoted attribute value instead.
